src/backend/binaryen/utils.ts

```diff
@@ -1816,23 +1869,128 @@ export namespace FunctionalFuncs {
                 }
                 break;
             }
-            default:
-                if (rightValueType.kind === ValueTypeKind.NUMBER) {
-                    res = operateF64F64ToDyn(
-                        module,
-                        leftValueRef,
-                        rightValueRef,
-                        opKind,
-                        true,
+            case ts.SyntaxKind.LessThanEqualsToken:
+            case ts.SyntaxKind.LessThanToken:
+            case ts.SyntaxKind.GreaterThanEqualsToken:
+            case ts.SyntaxKind.GreaterThanToken: {
+                let staticTypeKind = ValueTypeKind.UNDEFINED;
+                if (
+                    leftValueType.kind === ValueTypeKind.NUMBER ||
+                    rightValueType.kind === ValueTypeKind.NUMBER
+                ) {
+                    staticTypeKind = ValueTypeKind.NUMBER;
+                    if (leftValueType.kind === ValueTypeKind.NUMBER) {
+                        res = operateF64F64ToDyn(
+                            module,
+                            leftValueRef,
+                            rightValueRef,
+                            opKind,
+                            true,
+                            false,
+                        );
+                    } else {
+                        res = operateF64F64ToDyn(
+                            module,
+                            leftValueRef,
+                            rightValueRef,
+                            opKind,
+                            false,
+                            true,
+                        );
+                    }
+                } else if (
+                    leftValueType.kind === ValueTypeKind.STRING ||
+                    rightValueType.kind === ValueTypeKind.STRING
```

Comment:
I suggest that just remove the `else if (xx ===ValueTypeKind.STRING )` here, or just support these opKinds in `operateStrStrToDyn`, otherwise the error message `unreachable` will be very hard to find the root cause, and the throwed error is easy to find.